client/src/main/java/org/kakara/client/scenes/uicomponenets/ChatComponent.java

```diff
@@ -77,7 +78,7 @@ public ChatComponent(Font fontToUse, boolean alwaysShowHistory, Scene scene){
         this.historyRectangle.setVisible(alwaysShowHistory);
 
 
-        GameHandler.getInstance().getEventManager().registerHandler(this, scene);
+        GameHandler.getInstance().getCurrentScene().getEventManager().registerHandler(this);
```

Comment:
Scene is a parameter in the constructor. You should use `scene.getEventManager().registerHandler(this);`